pandas/core/arrays/masked.py

```diff
@@ -404,20 +453,17 @@ def isin(self, values) -> BooleanArray:  # type: ignore[override]
         from pandas.core.arrays import BooleanArray
 
         result = isin(self._data, values)
-        if self._hasna:
+        if self._mask is not None:
```

Comment:
There are multiple changes like this where `_mask is not None` was used instead of `_hasna`. `_hasna` seems more resistant to any future refactoring in how the mask works, but then we need an additional `assert self._mask is not None` afterwards anyway for `mypy`.